blueprints/draft_blueprint.py

```diff
@@ -20,25 +21,45 @@
 
 draft = Blueprint('draft', __name__)
 
+
 @draft.before_request
 def check_valid_login():
     """ Detect if user is logged in as the default - basically equivalent to `@login_required`
         TODO: force user to set their team name
     """
     # ensure user is logged in
     login_valid = current_user.is_authenticated
-    if (request.endpoint and 
-        'static' not in request.endpoint and 
-        not login_valid and 
-        not getattr(draft.view_functions.get(request.endpoint, {}), 'is_public', False) ) :
+    if (request.endpoint and
+            'static' not in request.endpoint and
+            not login_valid and
+            not getattr(draft.view_functions.get(request.endpoint, {}), 'is_public', False)):
         return render_template('login.html', next=request.endpoint)
 
+
 def public_endpoint(function):
     """ decorator to make public -- probably want to functools wraps this """
     function.is_public = True
     return function
 
 
+def generate_leaderboard(leaderboard_type):
+    differential_column_label = 'pick_differential'
+    sort_dir = asc(differential_column_label) if leaderboard_type == "worst" else desc(differential_column_label)
+
+    rows = db.session.query(Player.player_name_first,
+                            Player.player_name_last,
+                            User.username,
+                            (Selection.draftdraft_selection - Player.nfl_draft_pick).label(differential_column_label)) \
+        .join(Selection,
+              Selection.player_id == Player.player_id) \
+        .join(User,
+              Selection.selecting_team_id == User.user_id) \
+        .order_by(nulls_last(sort_dir)) \
+        .limit(3) \
```

Comment:
heads up i dont believe you need the "\". Generally you want to avoid them in python

```suggestion
                            (Selection.draftdraft_selection - Player.nfl_draft_pick).label(differential_column_label))
        .join(Selection,
              Selection.player_id == Player.player_id)
        .join(User,
              Selection.selecting_team_id == User.user_id)
        .order_by(nulls_last(sort_dir))
        .limit(3)
```